### PR TITLE
perf(robot-server, api-client): return summaries from `GET /runs`

### PR DESCRIPTION
**Pull request by mcous**
## Overview

Moving forward, the job of the `GET /runs` endpoint will be to give the client enough information to get all details via a call to `GET /runs/:run_id`.

## Changelog

- perf(robot-server, api-client): return summaries from `GET /runs`

## Review requests

App continues to function as before, and the robot may be a little less laggy.

This PR corrects some incorrect response schema typing in the api-client. This PR needs extra attention from apps&ui to make sure these changes are correct

## Risk assessment

Low

### CODE REVIEW

**Review by SyntaxColoring — APPROVED** [review]
Python changes and API changes look good. 👍

File: api-client/src/runs/types.ts [hunk 1]
```diff
@@ -49,21 +59,17 @@ export interface LabwareOffset {
   vector: VectorOffset
 }
 
-interface ResourceLink {
```
Comment (mcous) [inline api-client/src/runs/types.ts:52]: Duplicate type already defined at a higher level

File: api-client/src/runs/types.ts [hunk 2]
```diff
@@ -49,21 +59,17 @@ export interface LabwareOffset {
...
-  href: string
-  meta?: Partial<{ [key: string]: string | null | undefined }> | null
-}
-
-type ResourceLinks = Record<string, ResourceLink | string | null | undefined>
```
Comment (mcous) [inline api-client/src/runs/types.ts:57]: This part of the API strictly defines the keys of any links it returns, so we should do that on the client side, too

File: api-client/src/runs/types.ts [hunk 3]
```diff
@@ -49,21 +59,17 @@ export interface LabwareOffset {
...
-
 export interface Run {
   data: RunData
-  links?: ResourceLinks
```
Comment (mcous) [inline api-client/src/runs/types.ts:61]: I removed the `link` field from response types in this file that never return `links`

**Review by b-cooper — APPROVED** [review]
🦘

**Pull request merged**